dev.sparcles.SREngine.yml

```diff
@@ -0,0 +1,48 @@
+id: dev.sparcles.SREngine
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: srengine-run.sh
+
+build-options:
+  build-args:
+    - --share=network
+
+modules:
+   - name: srengine
+     buildsystem: cmake-ninja
+     config-opts:
+       - -DCMAKE_BUILD_TYPE=Debug
+       - -DSR_ENGINE_FLATPAK_BUILD=ON
+       - -DSR_ENGINE_EXTENDED_NAME=OFF
+
+     build-commands:
+       - cmake --build . -j ${FLATPAK_BUILDER_N_JOBS}
+       - install -Dm755 Engine/srengine /app/bin/srengine
+       - mkdir -p /app/share/SREngine
+       - cp -r ../Resources /app/share/SREngine
+     builddir: true
+     sources:
+       - type: git
+         url: https://github.com/SpaRcle-Studio/SREngine
+         branch: master
+
+   - name: script
+     buildsystem: simple
+     build-commands:
+       - install -Dm755 srengine-run.sh /app/bin/srengine-run.sh
+     sources:
+       - type: script
+         dest-filename: srengine-run.sh
+         commands:
+           - /app/bin/srengine -resources /app/share/SREngine/Resources
+
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --device=dri
+  - --device=input
```

Comment:
this device permission will NOT work 100% system that ship a stable version of flatpak.